lib/Parse/ParseExpr.cpp

```diff
@@ -2949,6 +3002,10 @@ ExprResult Parser::DeferredParseBoundsExpression(std::unique_ptr<CachedTokens> T
   ConsumeAnyToken();   // Skip past the current token to the new tokens.
   ExprResult Result = ParseBoundsExpression();
 
+  if (StartsRelativeBoundsExpression(Tok)) {
```

Comment:
This is not the right place to be parsing the relative alignment clause.  It needs to be parsed as part of ParseBoundsExpression, and only if comes after a 'bounds' expression.

I think a better place to parse the relative BoundsExpression is at line 2912 in ParseExpr.cpp (in the function ParseBoundsExpression).